### PR TITLE
feat: add multipart upload

### PR DESCRIPTION
**Pull request by TessaIO**
## Summary by CodeRabbit

- **New Features**
	- Introduced a new field for configuring maximum idle connections per host in S3 settings.
	- Added multipart upload functionality for handling larger file uploads efficiently.

- **Improvements**
	- Enhanced the upload process to automatically switch to multipart uploads based on file size.
	- Updated method for handling multipart upload options, improving error handling and logging.

### CODE REVIEW

File: src/infra/src/storage/remote.rs [hunk 1]
```diff
@@ -240,6 +247,9 @@ fn init_aws_config() -> object_store::Result<object_store::aws::AmazonS3> {
     if cfg.s3.feature_http2_only {
         opts = opts.with_http2_only();
     }
+    if cfg.s3.max_idle_per_host >= 0 {
+        opts = opts.with_pool_max_idle_per_host(cfg.s3.max_idle_per_host)
```
Comment (hengfeiyang) [inline src/infra/src/storage/remote.rs:251]: https://docs.rs/object_store/latest/object_store/struct.ClientOptions.html#method.with_pool_max_idle_per_host
it looks here we need `usize`, and use `-1` as default value is strange, the default value is `0`, no limit.

Comment (TessaIO) [reply src/infra/src/storage/remote.rs:251]: Done!

**Review by hengfeiyang — APPROVED** [review]

**Comment by hengfeiyang** [comment]
@TessaIO LGTM, thank you.

**Pull request merged**